File: stockie-back-end/app/models/stock.rb
class Stock < ApplicationRecord
  belongs_to :company
  belongs_to :user
  has_many :portfolio_stocks
  has_many :portfolios, through: :portfolio_stocks


  validates :ticker, :current_price, :min_price, :max_price, :company_id, presence: true
  validates :max_price, comparison: { greater_than: :min_price && :current_price }
  validates :min_price, comparison: { less_than: :max_price && :current_price }
end

File: stockie-front-end/src/models/Stock.ts
import {Company} from "./Company";
import {User} from "./User";

export interface Stock {
    id?: number;
    ticker: string;
    current_price: number;
    min_price: number;
    max_price: number;
    company_id?: number;
    company?: Company;
    user_id?: number;
    user?: User;
}
